src/common/args_stat_encoding.h

```
@@ -122,3 +152,28 @@ using saunafs_stat_t = struct stat;
 // On POSIX just call stat
 inline int stat_portable(const char *path, saunafs_stat_t *st) { return stat(path, st); }
 #endif
+
+// Reads the entire file into a std::string (UTF-8)
+inline bool read_file_to_string(const std::string &path, std::string &out) {
+#ifdef _WIN32
```

Couple of issues:

1. Reads can fail midway, even if the file can be opened (e.g EINTR, EIO etc.). They might be silently ignored or a exception throwed (not sure).
2. istreambuf_iterator is not really great for big files. Better for you to get the size of the file and then read everything at once.

Something like this:
```cpp

ifs.seekg(0, std::ios::end);
std::streampos size = ifs.tellg();
if (size < 0) { return false; }
ifs.seekg(0, std::ios::beg);

std::string result;
result.resize(static_cast<std::size_t>(size));
if (!size) {
    out.clear();
    return true;
}

ifs.read(result.data(), size);
if (ifs.bad() || (ifs.fail() && !ifs.eof())) {
    return false;
}
```

Also maybe the comment is a bit misleading, it doesn't actually do any checking/conversion on whether the file is UTF-8.

---

Done.